src/Raven.Server/Documents/Subscriptions/LiveSubscriptionPerformanceCollector.cs

```diff
@@ -0,0 +1,304 @@
+
+                var batchAggregators = new List<SubscriptionBatchStatsAggregator>(performance.Count);
+                
+                // 1. get 'ended' batches info
+                while (performance.TryTake(out SubscriptionBatchStatsAggregator stat))
+                {
+                    batchAggregators.Add(stat);
+                }
+                
+                // 2. get 'inProgress' batch info
+                using (Database.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                using (context.OpenReadTransaction())
+                {
+                    var inProgressConnection = Database.SubscriptionStorage.GetSubscriptionConnection(context, subscriptionName);
+                    var inProgressBatchStats = inProgressConnection?.Connection?.GetBatchPerformanceStats();
+                    if (inProgressBatchStats != null &&
+                        inProgressBatchStats.Completed == false &&
+                        batchAggregators.Contains(inProgressBatchStats) == false)
+                    {
+                        batchAggregators.Add(inProgressBatchStats);
+                    }
+                }
+                
+                // 3. add to results
+                batchPerformance.AddRange(batchAggregators.Select(x => x.ToBatchPerformanceLiveStatsWithDetails()));
+                var subscriptionItem = preparedStats.Find(x => x.TaskName == kvp.Key);
+                if (subscriptionItem != null)
+                {
+                    subscriptionItem.BatchPerformance = batchPerformance.ToArray();
+                }
+            }
+
+            return preparedStats;
+        }
+        
+        private class SubscriptionAndPerformanceConnectionStatsList
+            : HandlerAndPerformanceStatsList<SubscriptionStorage.SubscriptionGeneralDataAndStats, SubscriptionConnectionStatsAggregator>
+        {
+            public SubscriptionAndPerformanceConnectionStatsList(SubscriptionStorage.SubscriptionGeneralDataAndStats subscription) : base(subscription)
+            {
+            }
+        }
+        
+        private class SubscriptionAndPerformanceBatchStatsList
+            : HandlerAndPerformanceStatsList<SubscriptionStorage.SubscriptionGeneralDataAndStats, SubscriptionBatchStatsAggregator>
+        {
+            public SubscriptionAndPerformanceBatchStatsList(SubscriptionStorage.SubscriptionGeneralDataAndStats subscription) : base(subscription)
+            {
+            }
+        }
+        
+        private void OnEndConnection(SubscriptionConnection connectionThatEnded)
+        {
+            var subscriptionName = connectionThatEnded.SubscriptionState.SubscriptionName;
+
+            if (_perSubscriptionConnectionStats.TryGetValue(subscriptionName, out var subscriptionAndStats))
+            {
+                // add the 'ended connection' to the performance collection
```

Comment:
remove - it is obvious that you adding this in code

Reply:
done